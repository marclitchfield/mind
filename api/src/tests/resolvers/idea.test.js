import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Idea tests', () => {
  applySetup();

  test('create union', () => {
    return verifyMutations('Idea tests', 'create reaction', {
      mutations: [
        ({space}) => gql`mutation { Idea { subject:post_in_space(input:{sourceId:"${space.id}", title: "subject idea"}) { id } } }`,
        ({space}) => gql`mutation { Idea { response:post_in_space(input:{sourceId:"${space.id}", title: "response idea"}) { id } } }`,
        ({subject, response}) => gql`mutation { Idea { post_reaction(subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}" type:"answer") } }`,
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          ideas {
            title
            reactionsTo { type subject { title } response { title } }
            reactionsOf { type subject { title } response { title } }
          }
        }
      }`
    });
  });


});