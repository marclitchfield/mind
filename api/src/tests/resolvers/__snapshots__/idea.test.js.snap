// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea tests create union 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "ideas": Array [
      Object {
        "__typename": "Idea",
        "reactionsOf": Array [],
        "reactionsTo": Array [
          Object {
            "__typename": "Reaction",
            "response": Object {
              "__typename": "Idea",
              "title": "response idea",
            },
            "subject": Object {
              "__typename": "Idea",
              "title": "subject idea",
            },
            "type": "answer",
          },
        ],
        "title": "response idea",
      },
      Object {
        "__typename": "Idea",
        "reactionsOf": Array [
          Object {
            "__typename": "Reaction",
            "response": Object {
              "__typename": "Idea",
              "title": "response idea",
            },
            "subject": Object {
              "__typename": "Idea",
              "title": "subject idea",
            },
            "type": "answer",
          },
        ],
        "reactionsTo": Array [],
        "title": "subject idea",
      },
    ],
  },
}
`;
